commit 01b9136799e4c76e26c59dd266cd5435cccde534 (refs/changes/47/5647/4)
Author: Marsell Kukuljevic <marsell@joyent.com>
Date:   2019-02-28T18:20:07+00:00 (7 months ago)
    
    TRITON-1266 vmapi does not store disk UUID upon create
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/CHANGES.md b/CHANGES.md
index c1b86a6..e42959c 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -8,7 +8,7 @@
 
 ## Not yet released
 
-(nothing)
+- TRITON-1266 Allow a disk_uuid arg to `CreateDisk`.
 
 ## 12.2.0
 
diff --git a/lib/vmapi.js b/lib/vmapi.js
index e454031..c2b5dcf 100644
--- a/lib/vmapi.js
+++ b/lib/vmapi.js
@@ -788,6 +788,7 @@ VMAPI.prototype.removeNics = function (params, options, callback) {
  * @param {String} params.owner_uuid : Optional, the owner of the VM.
  * @param {Object} params.context : Optional, value to pass as x-context header
  * @param {String} params.pci_slot : Optional, PCI slot to be used by new disk
+ * @param {String} params.disk_uuid : Optional, UUID to be used by new disk
  * @param {Number | String} params.size : Size in MiB, or "remaining"
  * @param {Object} options : Request options.
  * @param {Function} callback : of the form f(err, job).
@@ -802,6 +803,9 @@ VMAPI.prototype.createDisk = function createDisk(params, options, callback) {
         throw new TypeError('UUID is required');
     if (params.pci_slot !== undefined && typeof (params.pci_slot) !== 'string')
         throw new TypeError('pci_slot must be a string');
+    if (params.disk_uuid !== undefined &&
+            typeof (params.disk_uuid) !== 'string')
+        throw new TypeError('disk_uuid must be a string');
     if (isNaN(+params.size) && params.size !== 'remaining')
         throw new TypeError('size is required (number or \'remaining\')');
     if (params.origin)
@@ -828,6 +832,7 @@ VMAPI.prototype.createDisk = function createDisk(params, options, callback) {
     }
 
     var args = {
+        disk_uuid: params.disk_uuid,
         pci_slot: params.pci_slot,
         size: params.size
     };
